src/ui/Folding/Folding.ts

```diff
@@ -414,12 +417,35 @@ export class Folding extends Component {
     const queryResults = data.results;
 
     const getResult: (result: IQueryResult) => IQueryResult = this.options.getResult || Folding.defaultGetResult;
-    queryResults.results = _.map(queryResults.results, getResult);
+    queryResults.results = map(queryResults.results, getResult);
+
+    if (this.options.rearrange) {
+      queryResults.results.forEach(result => {
+        result.childResults = sortBy(result.childResults, result => Utils.getFieldValue(result, this.options.rearrange.sort));
+        if (this.shouldBeReversed(result.childResults)) {
+          result.childResults = result.childResults.reverse();
+        }
+      });
+    }
+
     this.addLoadMoreHandler(<IQueryResult[]>queryResults.results, data.query);
   }
 
+  private shouldBeReversed(childResults: IQueryResult[]) {
+    if (this.options.rearrange.direction != 'descending') {
+      return false;
+    }
+    const anyChildResultHasNoValueToSortBy = any(childResults, childResult => {
+      return Utils.isNullOrUndefined(Utils.getFieldValue(childResult, this.options.rearrange.sort));
+    });
+    if (anyChildResultHasNoValueToSortBy) {
```

Comment:
Why should the order necessarily be reversed if all children have a `sortBy` value defined?

Edit: also, is there a UT for this case?

Reply:
Not necessarily. It's going to be reversed if we want it to be (direction-> descending) and that it's possible to do a valid sort operation on every child result (what this check here).

There's a UT for this, which set the rearrange option on folding to `@doesNotExist descending`, and check that the results do not get reverted or any error gets thrown.